compiler/rustc_middle/src/ty/mod.rs

```diff
@@ -1162,16 +1135,8 @@ pub enum PredicateAtom<'tcx> {
 
 impl<'tcx> Binder<PredicateAtom<'tcx>> {
     /// Wraps `self` with the given qualifier if this predicate has any unbound variables.
-    pub fn potentially_quantified(
-        self,
-        tcx: TyCtxt<'tcx>,
-        qualifier: impl FnOnce(Binder<PredicateAtom<'tcx>>) -> PredicateKind<'tcx>,
-    ) -> Predicate<'tcx> {
-        match self.no_bound_vars() {
-            Some(atom) => PredicateKind::Atom(atom),
-            None => qualifier(self),
-        }
-        .to_predicate(tcx)
+    pub fn potentially_quantified(self, tcx: TyCtxt<'tcx>) -> Predicate<'tcx> {
+        self.to_predicate(tcx)
```

Comment:
It's interesting, because this is actually a semantic change: before, we allowed binders here; now we don't (because of the `ToPredicate` impl). Maybe a bit surprisingly (though I'm actually glad since this is something I've been working towards), the only place this actually fails is in clippy. This basically means that we almost everywhere actually keeping track of bound vars (well, that they're there; we don't "lose" them to get them back later)

Reply:
Even better: the only reason the clippy tests were failing is because I used `binder.has_escaping_bound_vars` instead of `pred.has_escaping_bound_vars`